week2/src/backend/index.js

```diff
@@ -0,0 +1,120 @@
+
+const express = require("express");
+const app = express();
+const  mealsRouter = require("./routes/meals");
+const  reservationsRouter = require("./routes/reservations");
+const  reviewsRouter = require("./routes/reviews");
+
+
+app.use("/meals",function(req,res,next){
+  const meals =req.query;
+  const maxPrice = req.query.maxPrice;
+  const title=req.query.title ;
+  const createdAfter =req.query.createdAfter;
+  const limit =req.query.limit;
+  // Check if the object is empty
+  function isEmpty(obj) {
+    for (let key in obj) {
+      return false;
+    }
+    return true;
+  }
+  if (maxPrice || title || createdAfter|| limit ||  isEmpty(meals) ){
+    next()
+  }else{
+    res.send(404)
+  }
+
+})
+app.get("/meals", (req, res) => {
+  let correspondingMeals = mealsRouter;
+  const maxPrice = Number(req.query.maxPrice);
+  const title=req.query.title ;
+  const createdAfter =req.query.createdAfter;
+  const limit =Number(req.query.limit);
+// Get meals with a max price 
+  if (maxPrice){
+    correspondingMeals = mealsRouter.filter((meal)=>{
+                          return meal.price <= maxPrice ;
+                          })
+  }
+//  Get meals with  a specific title
+  if (title){
+    let reqTitle = new RegExp(title,"i")
+    correspondingMeals= correspondingMeals.filter((meal)=>{
+                          return meal.title.match(reqTitle);
+                          })
+}
+
+//Get meals that has been created after a specific date
+  if (createdAfter){
+    let reqCreatedAfter = Date.parse(createdAfter);
+    correspondingMeals = correspondingMeals.filter((meal)=>{
+                          mealCreatedAfter =  Date.parse(meal.createdAt) 
+                          return mealCreatedAfter - reqCreatedAfter > 0  ;
+                          })
+  }   
+ 
+// Get specific number of meals
+  if (limit){
+    correspondingMeals = mealsRouter.slice(0,limit);
```

Comment:
Hmm this looks a little weird, does this work?

Reply:
Yes, it is.Do you talk about limit =0 ? I have added :
if (limit == 0) {
    correspondingMeals = [];
    }  
??? what do you think ??

Reply:
The mealsRouter is an object that takes care of all the routing. So first off slicing that like an array seems weird. But also how can the first element of the mealsRouter (that is an object) be correspondingMeals?

Reply:
It is an array of objects,that's why I could use slice !